routes/projects/update.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * PUT /projects/:username/:projectName
+ * Creates a new project for the given user. A project object is sent in
+ * the form:
+ * {
+ *   name: <string> required,
+ *   username: <string> required,
+ *   isPublic: <boolean> required,
+ *   publishedAt: <dataTime> optional,
+ *   publishLink: <string> optional,
+ *   files: [ <FileObj1>, <FileObj2>... ] required
```

Comment:
re: updating files- new files should be created at `POST /files`, not at `PUT /projectname`. though we could implement a `PATCH /projectname` but that gets gross since the project should really only contain the foreign keys, not the actual file object representation...

tl;dr files should be made and updated at `POST/PUT /files`. they'll be associated with the project by foreign key. so when the `files` attribute is passed in a `PUT /projectname` it shouldn't be updating anything.